hotness/consumers.py

```diff
@@ -796,140 +733,230 @@ def get_dist_tag(self):
-                                                                                str(old_task_id),
-                                                                                str(task_id),
-                                                                                tmp)
-                                bug = bz
-                                text = text % (new_state, task_id)
-                                self.bugzilla.follow_up(text, bug)
-                            # Old sources failed or cancelled. We do not have nothing to do
-                            elif old_state in ('FAILED', 'CANCELED'):
-                                bug = None
-                                rh_stuff = None
-                            self.old_triggered_task_ids.pop(old_task_id)
-                            self.new_triggered_task_ids.pop(task_id)
-                        # Old sources has not been finished yet
-                        else:
-                            bug = None
-                            rh_stuff = None
-                    # Koji build failed and we have to attach build logs from rebase-helper
-                    else:
-                        if old_state is not None and old_state not in ('FAILED', 'CANCELED'):
-                            rh_stuff = self.buildsys.rebase_helper_checkers(new_version,
-                                                                            str(old_task_id),
-                                                                            str(task_id),
-                                                                            tmp)
-                            bug = bz
-                            self.old_triggered_task_ids.pop(old_task_id)
-                            self.new_triggered_task_ids.pop(task_id)
-                            text = text % (new_state, task_id)
-                            self.bugzilla.follow_up(text, bug)
+
+        old_task_id, new_task_id = self._find_tasks(task_id)
+
+        if old_task_id is None or new_task_id is None:
+            # Not a task triggered by rebase
+            return
+
+        old_task = self.old_triggered_task_ids[old_task_id]
+        new_task = self.new_triggered_task_ids[new_task_id]
+
+        if old_task[1] is None or new_task[1] is None:
+            # At least one task is still open
+            return
+
+        self._check_rebase(old_task_id, new_task_id)
+
+        self.old_triggered_task_ids.pop(old_task_id)
+        self.new_triggered_task_ids.pop(new_task_id)
+
+    def _rebase(self, package, version, upstream, bz):
+        """
+        Try to rebase the package with rebase-helper.
+
+        Runs rebase-helper and follows up on the bugzilla accordingly.
+
+        Returns True unless exception is raised during rebase.
+        """
+        tmp = tempfile.mkdtemp(prefix='thn-rh-', dir='/var/tmp')
+        try:
+            rh_stuff = None
+            try:
```

Comment:
This nested try block looks unnecessary and you should be able to remove it without affecting the behavior. You can also move the ``rh_stuff = None`` out of the try block so it's clear it's always going to be a variable in the ``except`` block as well.